src/NSubstitute/Core/WhenCalled.cs

```diff
@@ -70,4 +73,25 @@ public TException Throw<TException>() where TException : Exception, new()
     /// </summary>
     public void Throw(Func<CallInfo, Exception> createException) =>
         Do(ci => throw createException(ci));
+
+    /// <summary>
+    /// Throws the specified exception when called.
+    /// </summary>
+    [Obsolete(RecommendedThrowMethodToUseMessage)]
```

Comment:
**thought:** WDYT about not having `Obsolete` on this and just invisibly handling this user code error? Maybe even just code doc like `/// Prefer <see cref="Throw" /> for readability.` or `/// Did you mean <see cref="Throw" />?`.